### ISSUE TITLE
Implement adding new records

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar#197]
**Problem**

The user should be able to add new records to a table.

**Proposed solution**

We should support this in the frontend. See the design issue. 

**Additional context**

Record addition will use `POST` requests to `/api/v0/records/`

Blocked by design work in #193.

**Comment by pavish** [comment]
These comment from @zackkrida from https://github.com/centerofci/mathesar/pull/492#pullrequestreview-716263987 on locking other operations and showing save status, will be handled as part of this issue.

> There should be some kind of UI lock while a deletion is in progress. Currently many operations can be performed during a delete, which could be problematic in the future once row editing, creating, etc. are avaliable
> * The delete button should be disabled while a delete operation is in progress. If I throttle my network to a slow connection, I'm able to press the delete button while a deletion is in-progress, which sends the same DELETE requests again.
> * Currently, I can change sort order while a large (I tried 20 and 100 rows) delete is in progress. This removes the red background from the rows being deleted and still performs the sort.

> Confirmation on completion of deletion with a toast or other flash message would be ideal. It can be hard to tell just from the rows disappearing.
> Currently no error handling if a delete request fails (simulated by blocking the delete endpoint):

**Comment by pavish** [comment]
> There should be some kind of UI lock while a deletion is in progress. Currently many operations can be performed during a delete, which could be problematic in the future once row editing, creating, etc. are avaliable
> Confirmation on completion of deletion with a toast or other flash message would be ideal. It can be hard to tell just from the rows disappearing.

These operations require detailed UX definitions, which are beyond the scope of this issue. I will create a new issue for them.
